src/main/java/me/towdium/jecalculation/data/structure/RecordMath.java

```diff
@@ -69,7 +69,7 @@ public RecordMath(State state,
     public RecordMath(NBTTagCompound nbt) {
         state = State.values()[nbt.getInteger(KEY_STATE)];
         operator = Operator.values()[nbt.getInteger(KEY_OPERATOR)];
-        last = nbt.hasKey(KEY_LAST) ? new BigDecimal(nbt.getString(KEY_LAST)) : BigDecimal.ZERO;
+        last = nbt.hasKey(KEY_LAST) ? BigDecimal.valueOf(nbt.getDouble(KEY_LAST)) : BigDecimal.ZERO;
```

Comment:
Does the `getDouble` return a double value? If so, it may not work with `BigDecimal`. I'm not sure. In fact, I hardly ever use the math calculator.

Reply:
Sadly, you actually need to use math calculator once (i.e. calc result of 2 / 3) and save recipe. It will constantly crashes then, until you manually fix ```record.json``` :(

Yes, ```BigDecimal(double val)``` has unpredicted behaviour, as described in javadoc (i.e. ```new BigDecimal(0.1)```). Thats why I am using ```BigDecimal.valueOf(double val)``` instead. It is equal to ```BigDecimal(Double.toString(double val))```

In order to explain why I am replace ```nbt.getString``` with ```nbt.getDouble``` I need to explain what is actually going in here:

1. If number value has floating point in record.json it will be converted to ```NBTTagDouble``` or to ```NBTTagInt``` if not:
https://github.com/Towdium/JustEnoughCalculation/blob/6bdb1eba6c5c0a33b7fe0c168d4791bbeabf28de/src/main/java/me/towdium/jecalculation/utils/NBTJson.java#L95-L98

2.  If object has type ```NBTTagDouble``` and ```NBTTagDouble.getString``` is called then it will be converted with letter ```d``` at the end of the string, as described in forge sources:
![image](https://user-images.githubusercontent.com/23621659/149644043-c5d25950-eab6-415a-96c1-3dc21c60fb5d.png)
This wont happen for ```NBTTagInt``` object
![image](https://user-images.githubusercontent.com/23621659/149644551-f729072b-3a65-49a0-b295-38c2d5ec3f95.png)

3. i.e. ```"0.1"``` in json passes as ```"0.1d"``` into ```BigDecimal(string val)``` constructror, which will fail with ```NumberFormatException```
4. ```nbt.getDouble``` works fine with ```NBTTagDouble``` and ```NBTTagInt```, just tested it.

Reply:
Have you tested the latest version?  I fixed a bug about NBT and JSON in version 3.8.1.  The `toString` method of NBT object has never been used in the transformation process since version 3.8.1. (Or the `func_150286_g` is `toString`?)
https://github.com/Towdium/JustEnoughCalculation/blob/6bdb1eba6c5c0a33b7fe0c168d4791bbeabf28de/src/main/java/me/towdium/jecalculation/utils/NBTJson.java#L36-L39
Anyway, I tested the version 3.8.2, and it works fine.

Reply:
Just tested on fresh install with latest public version available and bug is there.  Test environment: MC 1.7.10 + Forge + JEC 0.8.2 + NEI + CodeChickenCore. 

1. Remove ```record.json```  from JEC config directory if that file exists.
2. Open Math Calculator, do 2 / 3 = 0.666(6) calc.
3. Open Craft Calculator, add new recipe (e.g. 1 chest = 8 wood planks) and save it.
4. Quit the game.
5. Launch the game. It won't load because of ```NumberFormatException```

Reply:
Oh, I understand it. You are right.